packages/esm-patient-banner-app/src/banner-tags/print-identifier-sticker.component.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button, ModalBody, ModalFooter, ModalHeader } from '@carbon/react';
+import styles from './print-identifier-sticker.scss';
+import { useReactToPrint } from 'react-to-print';
+import { Column } from '@carbon/react';
+import { Grid } from '@carbon/react';
+import { age, displayName, showSnackbar, useConfig } from '@openmrs/esm-framework';
+
+interface PrintIdentifierStickerProps {
+  patient: fhir.Patient;
+  closeModal: () => void;
+}
+
+const PrintIdentifierSticker: React.FC<PrintIdentifierStickerProps> = ({ patient, closeModal }) => {
+  const { t } = useTranslation();
+  const [isPrinting, setIsPrinting] = useState(false);
+  const contentToPrintRef = useRef(null);
+  const [printRetryCount, setprintRetryCount] = useState(0);
+  const { printIdentifierStickerFields, printIdentifierStickerSize, excludePatientIdentifierCodeTypes } = useConfig();
+
+  const headerTitle = t('patientIdentifierSticker', 'Patient Identifier Sticker');
+
+  const onBeforeGetContentResolve = useRef(null);
+
+  useEffect(() => {
+    if (isPrinting && onBeforeGetContentResolve.current) {
+      onBeforeGetContentResolve.current();
+    }
+  }, [isPrinting]);
```

Comment:
It would be best to move the DOM manipulation into here. Something like:

```ts
  const styleTag = useRef<HTMLStyleElement | null>(null);

  useEffect(() => {
    if (isPrinting) {
      const printStyles = `@media print { @page { size: ${printIdentifierStickerSize}; } }`;

      const style = document.createElement('style');
      style.appendChild(document.createTextNode(printStyles));

      styleTag.current = document.head.appendChild(style);

      onBeforeGetContentResolve.current?.();
    } else {
      if (styleTag.current !== null) {
        document.head.removeChild(styleTag.current);
        styleTag.current = null;
      }
    }
  }, [isPrinting, printIdentifierStickerSize]);
```